jax/experimental/jax2tf/tfjs_demo/Sketcher/sketcher.py

```diff
@@ -0,0 +1,278 @@
+from flax import linen as nn
+from flax.training import common_utils # type: ignore
+
+import jax # type: ignore
+from jax import lax
+from jax import numpy as jnp
+import numpy as np # type: ignore
+import tensorflow as tf # type: ignore
+from tensorflowjs.converters import convert_tf_saved_model # type: ignore
+
+from jax.config import config # type: ignore
+config.config_with_absl()
+
+from jax.experimental import jax2tf # type: ignore
+jax2tf.jax2tf.ENABLE_TF_CONVOLUTION = True
+
+TfVariable = Any
+
+class ExportWrapper(tf.train.Checkpoint):
+
+  def __init__(self, fn: Callable, params,
+               input_shape: Sequence[int],
+               batch_sizes: Sequence[int]):
+    """
+    Args:
+      fn: a function taking two arguments, the parameters and the batch of
+        images.
+      params: the parameters, as a list/tuple/dictionary of np.ndarray, to be
+        used as first argument for `fn`.
+     input_shape: the shape of the second argument of `fn` (except the batch size)
+     batch_sizes: a sequence of batch sizes for which to save the function.
+    """
+    super().__init__()
+    # Convert fn from JAX to TF.
+    self._fn = jax2tf.convert(fn, with_gradient=True)
+    # Create tf.Variables for the parameters.
+    self._params = tf.nest.map_structure(
+        lambda param: tf.Variable(param, trainable=False), params)
+    self.saved_signatures = {}
+    # TODO: hackish? We need a serving_default signature for the conversion
+    # to work properly, but we might be able to change that through some
+    # parameter of the converter.
+    self.saved_signatures["serving_default"] = (
+      self.__call__.get_concrete_function(
+          tf.TensorSpec([1] + list(input_shape), tf.float32)
+      ))
+
+    # Implement the interface from https://www.tensorflow.org/hub/reusable_saved_models
+    self.variables = tf.nest.flatten(self._params)
+    self.trainable_variables: List[TfVariable] = []
+
+    for batch_size in batch_sizes:
+      input_spec = tf.TensorSpec([batch_size] + list(input_shape), tf.float32)
+      cf = self.__call__.get_concrete_function(input_spec)
+      self.saved_signatures[f"serving_{batch_size}"] = cf
+
+  @tf.function(autograph=False)
+  def __call__(self, inputs):
+    outputs = self._fn(self._params, inputs)
+    return outputs
+
+def save_model(fn: Callable, params,
+               model_dir: str, *, input_shape: Sequence[int],
+               batch_sizes: Sequence[int]):
+  """Saves the SavedModel for a function"""
+  print(jax.make_jaxpr(fn)(params, np.ones([1] + list(input_shape))))
+  wrapper = ExportWrapper(
+    fn, params, input_shape,
+    batch_sizes=batch_sizes)
+  print(f"Saving the model to {model_dir}")
+  tf.saved_model.save(wrapper, model_dir, signatures=wrapper.saved_signatures)
+
+# The code below is an adaptation for Flax from the work published here:
+# https://blog.tensorflow.org/2018/07/train-model-in-tfkeras-with-colab-and-run-in-browser-tensorflowjs.html
+
+class FlaxQuickDraw:
```

Comment:
This class existed in the other notebook because I had two experiments that I wanted to keep separate. I think that here we can just move everything to the top-level